### PR TITLE
Router fix

### PR DESCRIPTION
**Pull request by rogerbutt**
i am a god. also all the code is bad. one webpack config fixed it.

### CODE REVIEW

File: client/src/components/SubmitCode.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React, { PropTypes, Component } from 'react'
-import { connect } from 'react-redux'
+import { connect } from 'react-redux';
+import { pushState } from 'redux-router';
```
Comment (rong2) [inline client/src/components/SubmitCode.js]: Is pushState used?

File: client/src/routes/routes.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import { Route, IndexRoute } from 'react-router';
+import App from '../containers/App';
+import Welcome from '../components/Welcome';
+import SubmitCode from '../components/SubmitCode';
+
+const routes = (
```
Comment (rong2) [inline client/src/routes/routes.js:7]: Does this need to be defined in its own file or can we keep everything in index.js? Just curious.

EDIT: Nevermind, I see it's being used separately in configureStore.js

File: client/src/containers/App.js [hunk 3]
```diff
@@ -3,14 +3,20 @@ import { bindActionCreators } from 'redux'
...
 import Welcome from '../components/Welcome'
 
+
 class App extends Component {
+    constructor(props) {
+        super(props);
+    }
 
     render() {
+        console.log(this.props);
```
Comment (rong2) [inline client/src/containers/App.js]: Let's take out the console.log()

Comment (rogerbutt) [inline client/src/containers/App.js]: sick review

File: client/src/components/SubmitCode.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import React, { PropTypes, Component } from 'react'
-import { connect } from 'react-redux'
+import { connect } from 'react-redux';
```
Comment (rong2) [inline client/src/components/SubmitCode.js:2]: Just for my edification, should we or should we not use semicolons?

Comment (rogerbutt) [inline client/src/components/SubmitCode.js:2]: I've been doing them, but i think we should actually just leave out the semicolons.
EDIT: JK do em.

**Pull request merged**